From bf63030982e9b2c555dab11b3f53bfd27a48f567 Mon Sep 17 00:00:00 2001
Message-Id: <bf63030982e9b2c555dab11b3f53bfd27a48f567.1562323529.git.aditya@kobol.io>
From: Aditya Prayoga <aditya@kobol.io>
Date: Fri, 5 Jul 2019 18:45:05 +0800
Subject: [PATCH] mvebu: helios4: Prepare for UART image

SPL failed to continue the booting process if ECC is enabled.
So disable the ECC and while at it also disable env storage and
autoboot.

Signed-off-by: Aditya Prayoga <aditya@kobol.io>
---
 board/kobol/helios4/helios4.c | 2 +-
 configs/helios4_defconfig     | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/board/kobol/helios4/helios4.c b/board/kobol/helios4/helios4.c
index 3c3592ecf5..780bef12a9 100644
--- a/board/kobol/helios4/helios4.c
+++ b/board/kobol/helios4/helios4.c
@@ -66,7 +66,7 @@ static struct mv_ddr_topology_map board_topology_map = {
 	    0, 0,			/* cas_wl cas_l */
 	    MV_DDR_TEMP_LOW,		/* temperature */
 	    MV_DDR_TIM_DEFAULT} },	/* timing */
-	BUS_MASK_32BIT_ECC,		/* Busses mask */
+	BUS_MASK_32BIT,		/* Busses mask */
 	MV_DDR_CFG_DEFAULT,		/* ddr configuration data source */
 	{ {0} },			/* raw spd data */
 	{0}				/* timing parameters */
diff --git a/configs/helios4_defconfig b/configs/helios4_defconfig
index 699ce0609f..3ccb613b8d 100644
--- a/configs/helios4_defconfig
+++ b/configs/helios4_defconfig
@@ -14,7 +14,7 @@ CONFIG_DEBUG_UART=y
 CONFIG_AHCI=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_NR_DRAM_BANKS=2
-CONFIG_BOOTDELAY=3
+CONFIG_BOOTDELAY=-1
 CONFIG_SYS_CONSOLE_INFO_QUIET=y
 # CONFIG_DISPLAY_BOARDINFO is not set
 CONFIG_DISPLAY_BOARDINFO_LATE=y
@@ -33,7 +33,6 @@ CONFIG_CMD_CACHE=y
 CONFIG_CMD_TIME=y
 # CONFIG_SPL_PARTITION_UUIDS is not set
 CONFIG_DEFAULT_DEVICE_TREE="armada-388-helios4"
-CONFIG_ENV_IS_IN_MMC=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_SPL_OF_TRANSLATE=y
 CONFIG_SCSI_AHCI=y
-- 
2.17.1

